ui_framework/src/components/outside_click_detector/outside_click_detector.js

```diff
@@ -0,0 +1,46 @@
+import {
+  cloneElement,
+  Component,
+} from 'react';
+import PropTypes from 'prop-types';
+
+export class KuiOutsideClickDetector extends Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    onOutsideClick: PropTypes.func.isRequired,
+  }
+
+  onClickOutside = event => {
+    if (!this.wrapperRef) {
+      return;
+    }
+
+    if (this.wrapperRef === event.target) {
+      return;
+    }
+
+    if (this.wrapperRef.contains(event.target)) {
+      return;
+    }
+
+    this.props.onOutsideClick();
+  }
+
+  componentDidMount() {
+    document.addEventListener('click', this.onClickOutside);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.onClickOutside);
+  }
+
+  render() {
+    const props = Object.assign({}, this.props.children.props, {
+      ref: node => {
+        this.wrapperRef = node;
+      },
+    });
+
+    return cloneElement(this.props.children, props);
```

Comment:
Whoops, sorry for missing the review request on this one. Looks good, though I think

`return React.Children.only(this.props.children, child => cloneElement(child, props));`

is still the best practice here (even if only for clearer error messaging to a future dev trying to use multiple children).